src/components/Map.js

```diff
@@ -19,13 +25,27 @@ const Map = ({
     ).then((res) => setNearbyPlaces(res));
   }, [mapProperties.center.lat, mapProperties.center.lng, setNearbyPlaces]);
 
+  function handleApiLoaded(map, maps) {
+    setMapsObj(map, maps);
+  }
+
+  // only runs if maps object is populated
+  if (!isObjEmpty(mapsObj)) {
+    const { map } = mapsObj;
+    map.addListener("center_changed", function () {
+      setUserHasPanned(true);
```

Comment:
Was looking forever to see where you set UserHasPanned to true 😅

Reply:
We enjoy treasure hunts here at Collab Lab. 🔎 💰